### ISSUE TITLE
Add `resettable` prop in `<AutocompleteInput>`

### ISSUE THREAD

**Issue by afilp** [marmelab/react-admin#2564]
**Is your feature request related to a problem? Please describe.**
It would be nice to have the `resettable` prop for the `<AutocompleteInput>` component too.

This way the user can clear the existing input fast. Now, we need to focus the input and press delete many times.

**Comment by fzaninotto** [comment]
Great idea. The core team won't work on it in the near future, so feel free to open a PR for it.

**Comment by marquesgabriel** [comment]
Hello again. Today i spent the morning trying to implement de resettable feature on the AutocompleteInput but i'm stuck in a problem. Here is what i did:

 - Replaced `<TextField/>` component for `<ResettableTextField/>` in `AutocompleteInput.js`
 - Added resettable boolean prop

The X icon button appeared! Nice!
but when its clicked triggers an error in Autosuggest.js: ```event.target is undefined```. This happens because ResettableTextField's method `handleClickClearButton()` passes an empty string as argument and the autosuggest component expects the default event argument for onChange()

What can I do then? Any ideas?

**Comment by EricTousignant** [comment]
Hello, I found this issue while trying to implement the same thing and have found a way to fix the error. I would like to open a PR but I'm not 100% sure how to proceed. Thanks.

**Comment by adibnaya** [comment]
> Hello, I found this issue while trying to implement the same thing and have found a way to fix the error. I would like to open a PR but I'm not 100% sure how to proceed. Thanks.

First thing is to put your code here so that others (me) might use it immediately and not have to wait, Thanks.

**Comment by atomicacorn** [comment]
I need a similar behaviour on the referenced AutocompleteInput where I clear the field automatically when the user has submitted the form (the form in question is a drawer on top of a custom chat component).

I'm able to clear the value of the field in redux form, but I seem to mess up the referenced resource's possibleValues: If the user reopens the drawer and uses the field, they can see their previous choice in the drop down, but when they submit a new query, the AutocompleteInput is not populated with the resulting data. If the user clears this search, and tries again, the field then behaves as expected.

Is there an action to reset the referenced data? I've taken a peek around the source, but couldn't find anything.

**Comment by JulienMattiussi** [comment]
For a fast solution, you can use a type="seach" on your Autocomplete input to display a native x button.

![image](https://user-images.githubusercontent.com/39904906/74038321-477acc80-49c0-11ea-970e-2aba10bbf02d.png)

``` js
<ReferenceInput
            label="Autocomplete with reset button"
            source="post_id"
            reference="posts"
            alwaysOn
>
            <AutocompleteInput
                optionText="title"
                options={{ type: 'search' }}
            />
</ReferenceInput>
```

**Comment by EricTousignant** [comment]
@JulienMattiussi , very interesting, will try that! :)

**Comment by EricTousignant** [comment]
@JulienMattiussi , it's a good middle of the road approach, but if you click the X, then click outside, the previous value would be left behind. It's not exactly "clearing" the input as it should.

**Comment by melvynhills** [comment]
There's an `Autocomplete` component in the [@material-ui/lab package](https://material-ui.com/api/autocomplete/) that is pretty stable already. It has many options and is overall better than the current Downshift component in RA.

I started integrating it into a custom RA input component and it seems to work quite well. It's pretty easy to integrate with https://marmelab.com/react-admin/Inputs.html#useinput-hook.

**Comment by capJavert** [comment]
Hello, is there any interest for developing this? @fzaninotto

**Comment by RobertLowe** [comment]
Would also like this, bumping issue

**Comment by fzaninotto** [comment]
Fixed by #5396
